src/python/WMCore/WMRuntime/Watchdog.py

```diff
@@ -78,6 +81,35 @@ def setupMonitors(self, task, wmbsJob):
                 #This should be a config section
                 monitorArgs = getattr(task.data.watchdog, monitor)
                 args = monitorArgs.dictionary_()
+                # Scale resources according to the HTCondor runtime environment.
+                origCores = 1
+                try:
+                    origCores = int(task.data.application.multicore.numberOfCores)
+                except AttributeError:
+                    print("No value for numberOfCores. Not setting")
+                resources = {'cores': origCores}
+                origMaxRSS = args.get('maxRSS')
+                if origMaxRSS:
+                    resources['memory'] = origMaxRSS
+                # Actually parses the HTCondor runtime
+                resizeResources(resources)
+                # HTCondor doesn't explicitly scale VSize; hence we scale it
+                # according to the scaling of maxRSS between the workflow and runtime.
+                if origMaxRSS and ('maxVSize' in args):
+                    args['maxVsize'] *= resources['memory'] / origMaxRSS
+                # Scale our timeouts according to the scaling of the cores.  *NOTE*
+                # the job ad contains the real scaling here - unfortunately, it uses
+                # ClassAd expressions and we can only parse values.  Something for the
+                # future.
+                coreScaling = resources['cores'] / origCores
```

Comment:
I'm not sure we should scale the timeout values. A 4 core job meant to run in single core mode won't run 4 times faster, I think it would stay between 3.0 and 3.5 times faster, but depending on the step it could be even worse.